### ISSUE TITLE
TypeError: __init__() got an unexpected keyword argument 'feature_names'

### ISSUE THREAD

**Issue by netoolii** [shap/shap#1743]
Trying to instantiate Explainer and reproduce the following code at  [https://github.com/slundberg/shap/blob/master/notebooks/text_examples/sentiment_analysis/Emotion%20Data%20Multiclass%20Text%20Explanation%20Demo.ipynb](https://github.com/slundberg/shap/blob/master/notebooks/text_examples/sentiment_analysis/Emotion%20Data%20Multiclass%20Text%20Explanation%20Demo.ipynb)

I got this error:

```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-49-4f411f5b2215> in <module>
----> 1 explainer = shap.Explainer(f,e,output_names=predictor_explainable.tg_mapping.values())

~/.conda/envs/exp/lib/python3.7/site-packages/shap/explainers/_explainer.py in __init__(self, model, masker, link, algorithm, output_names, feature_names, **kwargs)
    136             elif algorithm == "permutation":
    137                 self.__class__ = explainers.Permutation
--> 138                 explainers.Permutation.__init__(self, model, self.masker, link=self.link, feature_names=self.feature_names, **kwargs)
    139             elif algorithm == "partition":
    140                 self.__class__ = explainers.Partition

TypeError: __init__() got an unexpected keyword argument 'feature_names'
```

Searching for this file I found the constructor:

```
/home/mycomputer/.conda/envs/exp/lib/python3.7/site-packages/shap/explainers/_permutation.py
class Permutation(Explainer):
    """ This method approximates the Shapley values by iterating through permutations of the inputs.

    This is a model agnostic explainer that gurantees local accuracy (additivity) by iterating completely
    through an entire permutatation of the features in both forward and reverse directions. If we do this
    once, then we get the exact SHAP values for models with up to second order interaction effects. We can
    iterate this many times over many random permutations to get better SHAP value estimates for models
    with higher order interactions. This sequential ordering formulation also allows for easy reuse of
    model evaluations and the ability to effciently avoid evaluating the model when the background values
    for a feature are the same as the current input value. We can also account for hierarchial data
    structures with partition trees, something not currently implemented for KernalExplainer or SamplingExplainer.
    """

    def __init__(self, model, masker, link=links.identity):
        """ Build an explainers.Permutation object for the given model using the given masker object.

        Parameters
        ----------
        model : function
            A callable python object that executes the model given a set of input data samples.

        masker : function or numpy.array or pandas.DataFrame
            A callable python object used to "mask" out hidden features of the form `masker(x, mask)`.
            It takes a single input sample and a binary mask and returns a matrix of masked samples. These
            masked samples are evaluated using the model function and the outputs are then averaged.
            As a shortcut for the standard masking using by SHAP you can pass a background data matrix
            instead of a function and that matrix will be used for masking. To use a clustering
            game structure you can pass a shap.maksers.Tabular(data, clustering=\"correlation\") object.
        """
        super(Permutation, self).__init__(model, masker, link=link)
```

if you look, the Permutation and the other classes do not have `feature_names` as an argument

**Libraries installed**

```
Python 3.7.4 


Package                           Version       
--------------------------------- --------------
absl-py                           0.9.0         
aiohttp                           3.6.2         
appdirs                           1.4.4         
APScheduler                       3.6.3         
asn1crypto                        1.3.0         
astor                             0.8.1         
astunparse                        1.6.3         
async-timeout                     3.0.1         
attrs                             19.3.0        
autokeras                         1.0.2         
autopep8                          1.5.4         
backcall                          0.1.0         
beautifulsoup4                    4.6.0         
bleach                            3.1.4         
blinker                           1.4           
boto                              2.49.0        
boto3                             1.12.39       
botocore                          1.15.39       
cachetools                        3.1.1         
certifi                           2020.4.5.1    
cffi                              1.14.0        
chardet                           3.0.4         
cld2-cffi                         0.1.4         
click                             7.1.2         
cloudpickle                       1.4.1         
colorama                          0.4.3         
conda                             4.8.1         
conda-package-handling            1.6.0         
configparser                      5.0.1         
cryptography                      2.8           
cycler                            0.10.0        
dart-trec-commons                 1.0.0         
dart-trec-learner                 1.0.0         
dask                              2.16.0        
deap                              1.3.1         
decorator                         4.4.2         
defusedxml                        0.6.0         
dill                              0.3.1.1       
distlib                           0.3.0         
distributed                       2.16.0        
dnspython                         1.16.0        
docutils                          0.15.2        
entrypoints                       0.3           
fasttext                          0.9.2         
filelock                          3.0.12        
flake8                            3.8.4         
future                            0.18.2        
gast                              0.3.3         
gensim                            3.8.0         
google                            2.0.3         
google-api-core                   1.16.0        
google-auth                       1.13.1        
google-auth-oauthlib              0.4.1         
google-cloud-core                 1.3.0         
google-cloud-storage              1.27.0        
google-pasta                      0.2.0         
google-resumable-media            0.5.0         
googleapis-common-protos          1.51.0        
grpcio                            1.29.0        
h5py                              2.10.0        
HeapDict                          1.0.1         
idna                              2.9           
imageio                           2.9.0         
iml                               0.6.2         
importlib-metadata                1.6.0         
info-gain                         1.0.1         
iniconfig                         1.1.1         
ipykernel                         5.2.1         
ipython                           7.13.0        
ipython-genutils                  0.2.0         
ipywidgets                        7.5.1         
jedi                              0.17.0        
Jinja2                            2.11.2        
jmespath                          0.9.4         
joblib                            0.14.1        
JPype1                            0.7.3         
json5                             0.9.0         
jsonschema                        3.2.0         
jupyter-client                    6.1.3         
jupyter-contrib-core              0.3.3         
jupyter-contrib-nbextensions      0.5.1         
jupyter-core                      4.6.3         
jupyter-highlight-selected-word   0.2.0         
jupyter-latex-envs                1.4.6         
jupyter-nbextensions-configurator 0.4.1         
jupyterlab                        2.1.1         
jupyterlab-server                 1.1.1         
jupyterthemes                     0.20.0        
Keras                             2.3.1         
Keras-Applications                1.0.8         
Keras-Preprocessing               1.1.2         
keras-tuner                       1.0.1         
kiwisolver                        1.1.0         
konlpy                            0.5.2         
lesscpy                           0.14.0        
lime                              0.2.0.1       
llvmlite                          0.35.0        
lxml                              4.5.0         
Markdown                          3.2.2         
MarkupSafe                        1.1.1         
matplotlib                        3.1.3         
mccabe                            0.6.1         
mecab-python                      0.996-ko-0.9.2
minio                             6.0.0         
mistune                           0.8.4         
mkl-fft                           1.0.15        
mkl-random                        1.1.0         
mkl-service                       2.3.0         
modin                             0.7.3         
msgpack                           1.0.0         
multidict                         4.7.5         
mysql-connector-python            8.0.19        
nbconvert                         5.6.1         
nbformat                          5.0.6         
networkx                          2.5           
nltk                              3.5           
notebook                          5.7.8         
numba                             0.52.0        
numpy                             1.18.5        
oauthlib                          3.1.0         
opt-einsum                        3.2.1         
packaging                         20.3          
pandarallel                       1.4.8         
pandas                            1.1.2         
pandocfilters                     1.4.2         
parso                             0.7.0         
pexpect                           4.8.0         
pickleshare                       0.7.5         
Pillow                            7.2.0         
pip                               19.2.3        
pipenv                            2020.6.2      
pipfile                           0.0.2         
pluggy                            0.13.1        
ply                               3.11          
prometheus-client                 0.7.1         
prompt-toolkit                    3.0.5         
protobuf                          3.11.3        
psutil                            5.7.0         
ptyprocess                        0.6.0         
py                                1.9.0         
py-spy                            0.3.3         
py4j                              0.10.7        
pyarrow                           0.16.0        
pyasn1                            0.4.8         
pyasn1-modules                    0.2.7         
pybind11                          2.5.0         
pycld2                            0.41          
pycld3                            0.20          
pycodestyle                       2.6.0         
pycosat                           0.6.3         
pycparser                         2.20          
pyflakes                          2.2.0         
Pygments                          2.6.1         
PyJWT                             1.7.1         
pymongo                           3.11.0        
PyMySQL                           0.9.3         
pyOpenSSL                         19.1.0        
pyparsing                         2.4.6         
pyrsistent                        0.16.0        
PySocks                           1.7.1         
pyspark                           2.4.4         
pytest                            6.1.1         
python-dateutil                   2.8.1         
pytz                              2019.3        
PyWavelets                        1.1.1         
PyYAML                            5.3.1         
pyzmq                             19.0.0        
ray                               0.8.4         
redis                             3.5.1         
regex                             2020.11.13    
requests                          2.23.0        
requests-oauthlib                 1.3.0         
rsa                               4.0           
ruamel-yaml                       0.15.87       
s3transfer                        0.3.3         
scikit-image                      0.18.0        
scikit-learn                      0.23.2        
scipy                             1.4.1         
seaborn                           0.10.1        
Send2Trash                        1.5.0         
setuptools                        51.1.2        
shap                              0.37.0        
sip                               4.19.13       
six                               1.14.0        
slicer                            0.0.3         
smart-open                        1.11.1        
sortedcontainers                  2.1.0         
spark-nlp                         2.5.1         
SQLAlchemy                        1.3.16        
stopit                            1.1.2         
tabulate                          0.8.7         
tblib                             1.6.0         
tensorboard                       2.4.0         
tensorboard-plugin-wit            1.7.0         
tensorflow                        2.3.1         
tensorflow-estimator              2.3.0         
tensorflow-hub                    0.10.0        
tensorflow-text                   2.3.0         
termcolor                         1.1.0         
terminado                         0.8.3         
terminaltables                    3.1.0         
testpath                          0.4.4         
threadpoolctl                     2.1.0         
tifffile                          2020.12.8     
toml                              0.10.1        
toolz                             0.10.0        
tornado                           6.0.4         
TPOT                              0.11.2        
tqdm                              4.36.1        
traitlets                         4.3.3         
tweepy                            3.8.0         
twkorean                          0.1.5         
tzlocal                           2.1           
update-checker                    0.17          
urllib3                           1.25.8        
virtualenv                        20.0.24       
virtualenv-clone                  0.5.4         
wcwidth                           0.1.9         
webencodings                      0.5.1         
Werkzeug                          1.0.1         
wheel                             0.34.2        
widgetsnbextension                3.5.1         
wrapt                             1.12.1        
xgboost                           1.1.0         
yapf                              0.30.0        
yarl                              1.4.2         
zict                              2.0.0         
zipp                              3.1.0   
```

**OS information**

```
NAME="CentOS Linux"
VERSION="7 (Core)"
ID="centos"
ID_LIKE="rhel fedora"
VERSION_ID="7"
PRETTY_NAME="CentOS Linux 7 (Core)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:centos:centos:7"
HOME_URL="https://www.centos.org/"
BUG_REPORT_URL="https://bugs.centos.org/"

CENTOS_MANTISBT_PROJECT="CentOS-7"
CENTOS_MANTISBT_PROJECT_VERSION="7"
REDHAT_SUPPORT_PRODUCT="centos"
REDHAT_SUPPORT_PRODUCT_VERSION="7"
```

**Comment by Timliuzhaolu** [comment]
I am having a similar issue.

**Comment by BrunoBarre** [comment]
Hi, I am having a similar issue. The following tutoriel from the SHAP documentation fails : [https://shap.readthedocs.io/en/latest/example_notebooks/image_examples/image_classification/Multi-class%20ResNet50%20on%20ImageNet%20%28TensorFlow%29.html](https://shap.readthedocs.io/en/latest/example_notebooks/image_examples/image_classification/Multi-class%20ResNet50%20on%20ImageNet%20%28TensorFlow%29.html).

It fails at the `explainer = shap.Explainer(f, masker, output_names=class_names)` line, saying :
![image](https://user-images.githubusercontent.com/50590102/106765537-313a3780-6639-11eb-9ee6-20cd6068d72e.png)


I have shap version 0.37.0 installed.

Thanks in advanced

**Comment by maxwellreynolds** [comment]
Looks like this is fixed in version 0.38

**Comment by BrunoBarre** [comment]
Ok cool, but conda version is still 0.37.0 (as in conda-forge [https://anaconda.org/conda-forge/shap](https://anaconda.org/conda-forge/shap)).
So this will be good once 0.38 version is available via conda. Thanks
